### PR TITLE
Add the TypeScript code and docs for Chapter of Linear Search

### PR DESCRIPTION
**Pull request by danielsss**
added: linear_search.ts and arrToLinkedList function
updated: linear search docs for Typescript

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# EditorConfig is awesome: https://EditorConfig.org
```
Comment (krahets) [inline .editorconfig:1]: Is this an editor-related file or a language-related file?

**Review by krahets — COMMENTED** [review]
Thanks! Waiting the final review by @justin-tse

File: .editorconfig (hunk 1 above)
Comment (danielsss) [reply .editorconfig:1]: It's just an editor-related file.

Comment (krahets) [reply .editorconfig:1]: I see. Please remove it. We should not keep any editor-related files in the repo.

Comment (danielsss) [reply .editorconfig]: done

**Review by justin-tse — APPROVED** [review]
Thank @danielsss for the clean codes!

**Review by krahets — APPROVED** [review]

**Pull request merged**